### PR TITLE
order impl

### PR DESCRIPTION
**Pull request by artemsa223**

### CODE REVIEW

File: src/main/java/com/example/demo/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+@Tag(name = "Order manager", description = "Endpoints for managing orders")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @Operation(summary = "Create order")
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (Vasul-Pankiv) [inline src/main/java/com/example/demo/controller/OrderController.java]: it is not necessary to check for USER role if you have only USER and ADMIN roles

File: src/main/java/com/example/demo/mapper/CartItemMapper.java [hunk 2]
```diff
@@ -8,14 +8,12 @@
...
 import org.mapstruct.MappingTarget;
-import org.mapstruct.Mappings;
 
 @Mapper(config = MapperConfig.class, uses = BookMapper.class)
 public interface CartItemMapper {
-    @Mappings({
-            @Mapping(target = "bookId", source = "book.id"),
-            @Mapping(target = "bookTitle", source = "book.title")
-    })
+
```
Comment (Vasul-Pankiv) [inline src/main/java/com/example/demo/mapper/CartItemMapper.java]: remove empty line

File: src/main/java/com/example/demo/mapper/ShoppingCartMapper.java [hunk 3]
```diff
@@ -5,19 +5,15 @@
...
 import org.mapstruct.Mapping;
-import org.mapstruct.Mappings;
 
 @Mapper(config = MapperConfig.class, uses = CartItemMapper.class)
 public interface ShoppingCartMapper {
-    @Mappings({
-            @Mapping(source = "user.id", target = "userId"),
-            @Mapping(source = "cartItems", target = "cartItems")
-    })
+
```
Comment (Vasul-Pankiv) [inline src/main/java/com/example/demo/mapper/ShoppingCartMapper.java]: empty line

File: src/main/java/com/example/demo/service/shoppingcart/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -25,10 +25,15 @@ public class ShoppingCartServiceImpl implements ShoppingCartService {
...
-    public ShoppingCartDto getShoppingCartByUserId(Long id) {
+    public ShoppingCartDto getShoppingCartDtoByUserId(Long id) {
         return cartMapper.toDto(getCartByUserId(id));
     }
 
+    @Override
+    public ShoppingCart getShoppingCartByUserId(Long id) {
+        return getCartByUserId(id);
+    }
+
```
Comment (Vasul-Pankiv) [inline src/main/java/com/example/demo/service/shoppingcart/ShoppingCartServiceImpl.java]: why you need this method, use getCartByUserId directly, It is 2 identical methods but better to use getShoppingCartByUserId name

File: src/main/java/com/example/demo/service/order/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+    @Transactional
+    @Override
+    public OrderDto create(Long userId, CreateOrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartService.getShoppingCartByUserId(userId);
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new IllegalStateException("Empty shopping cart!");
+        }
+        Order order = createOrder(userId, requestDto);
+        order = orderRepository.save(order);
```
Comment (Vasul-Pankiv) [inline src/main/java/com/example/demo/service/order/OrderServiceImpl.java]: why are you saving an order here, it is redundant

File: src/main/resources/db/changelog/changes/16-insert-cartitems.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+            columns:
+              - column:
+                  name: shopping_cart_id
+                  value: "1"
+              - column:
+                  name: book_id
+                  value: "1"
+              - column:
+                  name: quantity
+                  value: "3"
```
Comment (Vasul-Pankiv) [inline src/main/resources/db/changelog/changes/16-insert-cartitems.yaml]: File should end with empty line

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

**Review by koval-kn — APPROVED** [review]
looks good

**Pull request merged**